index.js

```diff
@@ -17,32 +17,24 @@ const pThrottle = ({limit, interval}) => {
 	}
 
 	const queue = new Map();
-
-	let currentTick = 0;
-	let activeCount = 0;
+	const ticks = [];
 
 	return function_ => {
 		const throttled = function (...args) {
 			let timeout;
 			return new Promise((resolve, reject) => {
 				const execute = () => {
+					ticks.pop();
 					resolve(function_.apply(this, args));
 					queue.delete(timeout);
 				};
 
 				const now = Date.now();
+				const executeAt = ticks.length < limit ? now : ticks[limit - 1] + interval;
```

Comment:
```suggestion
				const runDate = ticks.length < limit ? now : ticks[limit - 1] + interval;
```

might be a clearer name.

Reply:
I changed the algorithm a bit so this is no longer relevant.